Wrappers/Python/cil/optimisation/functions/L1Sparsity.py

```diff
@@ -0,0 +1,152 @@
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Authors:
+# CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
+
+from cil.optimisation.functions import Function, L1Norm
+import warnings
+
+class L1Sparsity(Function):
+
+    r"""L1Sparsity function
+
+    Calculates the following cases, depending on if the optional parameter `weight`  or data `b` is passed. For `weight=None`: 
+
+
+    a) .. math:: F(x) = ||Qx||_{1}
+    b) .. math:: F(x) = ||Qx - b||_{1}
+
+    In the weighted case, `weight` = :math:`w` is an array of positive weights.
+
+    a) .. math:: F(x) = ||Qx||_{L^1(w)}
+    b) .. math:: F(x) = ||Qx - b||_{L^1(w)}
+
+    with :math:`||x||_{L^1(w)} = || x \cdot w||_1 = \sum_{i=1}^{n} |x_i| w_i`.
+    
+    In all cases :math:`Q` is an orthogonal operator. 
+    
+    Parameters
+    ---------
+    Q: orthogonal Operator 
+        Note that for the correct calculation of the proximal the provided operator must be orthogonal 
+    b : Data, DataContainer, default is None 
+    weight: array, optional, default=None
+        positive weight array matching the size of the range of operator :math:`Q`.
+    """
+
+    def __init__(self, Q, b=None, weight=None):
+        '''creator
+        '''
+
+        if not Q.is_orthogonal(): 
+            warnings.warn(
+                f"Invalid operator: `{Q}`. L1Sparsity is only defined for orthogonal operators!", UserWarning)
+
+        super(L1Sparsity, self).__init__()
+        self.Q = Q
+
+        self.l1norm = L1Norm(b=b, weight=weight)
+
+    def __call__(self, x):
+        r"""Returns the value of the L1Sparsity function at x.
+
+        Consider the following cases:
+
+        a) .. math:: F(x) = ||Qx||_{1}
+        b) .. math:: F(x) = ||Qx - b||_{1}
+
+        In the weighted case, `weight` = :math:`w` is an array of positive weights.
+
+        a) .. math:: F(x) = ||Qx||_{L^1(w)}
+        b) .. math:: F(x) = ||Qx - b||_{L^1(w)}
+
+        with :math:`||x||_{L^1(w)} = || x w||_1 = \sum_{i=1}^{n} |x_i| w_i`.
+            
+        """
+        y = self.Q.direct(x)
+        return self.l1norm(y)
+
+    def convex_conjugate(self, x):
+        r"""Returns the value of the convex conjugate of the L1Sparsity function at x.
+        Here, we need to use the convex conjugate of L1Sparsity, which is the Indicator of the unit 
+        :math:`\ell^{\infty}` norm on the operator domain. (Since Q is a basis of L^2).
```

Comment:
Clarified this part and weakened the condition on the operator. Orthogonal operators are of course bijective.
```suggestion
        :math:`\ell^{\infty}` norm on the range of the (bijective) operator Q.
```